### PR TITLE
Test sf4

### PR DESCRIPTION
**Pull request by greg0ire**
~~Do not merge, this is a proof PR.~~ the dev-kit PR has been merged, so this should be mergeable too.

### CODE REVIEW

**Comment by greg0ire** [comment]
This can be merged now, but I need to fix a test on SonataCore for the sf3.4 build to be green.

**Comment by greg0ire** [comment]
> This can be merged now, but I need to fix a test on SonataCore for the sf3.4 build to be green.

See https://github.com/sonata-project/SonataCoreBundle/pull/457

@Soullivaneuh ideally, merge that one before, then release, and let me bump the version constraint in this PR. If you think this is going to be slow, then please merge it now instead.

**Comment by greg0ire** [comment]
Green: https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/297337761

**Comment by greg0ire** [comment]
This allows testing sf4. Currently it breaks because I need to do yet another correction on the abstract widget tests case, because I got no deprecation warning regarding `initRuntime`, which is removed in sf4. I'd rather do that and fix other sf4-related deprecations, but for that I need https://github.com/sonata-project/SonataCoreBundle/pull/456. This can be merged now, at least it will fix the deprecations build, and make sf4 testable by Travis.

File: Tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php [hunk 1]
```diff
@@ -479,14 +490,20 @@ public function testProcessAbstractAdminServiceInServiceDefinition()
             ->setArguments(['', 'Sonata\AdminBundle\Tests\DependencyInjection\Post', ''])
             ->setAbstract(true);
 
-        $adminDefinition = new DefinitionDecorator('sonata_abstract_post_admin');
+        $adminDefinition = class_exists(ChildDefinition::class) ?
```
Comment (jlamur) [inline Tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php]: This is a BC trick for SF > 3.3, please add a NEXT_MAJOR comment.

File: Tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php [hunk 2]
```diff
@@ -479,14 +490,20 @@ public function testProcessAbstractAdminServiceInServiceDefinition()
...
+            new ChildDefinition('sonata_abstract_post_admin') :
+            new DefinitionDecorator('sonata_abstract_post_admin');
         $adminDefinition
+            ->setPublic(true)
             ->setClass('Sonata\AdminBundle\Tests\DependencyInjection\MockAbstractServiceAdmin')
             ->setArguments([0 => 'extra_argument_1'])
             ->addTag('sonata.admin', ['group' => 'sonata_post_one_group', 'manager_type' => 'orm']);
 
-        $adminTwoDefinition = new DefinitionDecorator('sonata_abstract_post_admin');
+        $adminTwoDefinition = class_exists(ChildDefinition::class) ?
```
Comment (jlamur) [inline Tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php]: Same here.

File: Tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php [hunk 3]
```diff
@@ -194,11 +196,20 @@ function ($element) {
...
         ));
-        $this->assertSame(
-            'sonata.admin.route.path_info_slashes',
-            (string) $articleRouteBuilderMethodCall[1][0],
-            'The article admin uses the odm, and should therefore use the path_info_slashes router.'
-        );
+        $definitionOrReference = $articleRouteBuilderMethodCall[1][0];
+        if ($definitionOrReference instanceof Definition) {
+            $this->assertSame(
+                'Sonata\DoctrinePHPCRAdminBundle\Route\PathInfoBuilderSlashes',
```
Comment (Soullivaneuh) [inline Tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php]: `::class` notation.

**Comment by OskarStark** [comment]
No travis test with SF 4.0 beta?

**Comment by greg0ire** [comment]
@OskarStark it's called dev-master ;)

**Pull request merged**